addons/account/models/account_tax_report.py

```diff
@@ -227,12 +226,23 @@ def unlink(self):
     def _remove_tags_used_only_by_self(self):
         """ Deletes and removes from taxes and move lines all the
         tags from the provided tax report lines that are not linked
-        to any other tax report lines.
+        to any other tax report lines nor moves.
+        If there is still a link with moves, it will be archived instead
+        to avoid loosing data.
         """
         all_tags = self.mapped('tag_ids')
         tags_to_unlink = all_tags.filtered(lambda x: not (x.tax_report_line_ids - self))
         self.write({'tag_ids': [(3, tag.id, 0) for tag in tags_to_unlink]})
-        self._delete_tags_from_taxes(tags_to_unlink.ids)
+
+        move_with_archive_tag = self.env['account.move.line'].sudo().search([('tax_tag_ids', 'in', tags_to_unlink.ids)], limit=1)
```

Comment:
```suggestion
        aml_using_tags = self.env['account.move.line'].sudo().search([('tax_tag_ids', 'in', tags_to_unlink.ids)], limit=1)
```

Calling it move gives the impression it's an account.move to the reader, while it's not => move_line or aml (3-letters variable are generally a bad idea, but this abbreviation is widely used).
Name was unclear also "aml with archive tag" does not really reflect the use of this variable